plutus-benchmark/validation/Common.hs

```diff
@@ -73,30 +73,33 @@ withAnyPrefixFrom :: [String] -> [String] -> [String]
 l `withAnyPrefixFrom` ps =
     concatMap (\p -> filter (isPrefixOf p) l) ps
 
-loadFlat :: FilePath -> IO Term
-loadFlat file = do
-  contents <- BSL.fromStrict <$> BS.readFile file
-  case unflat contents of
-    Left e  -> errorWithoutStackTrace $ "Flat deserialisation failure for " ++ file ++ ": " ++ show e
-    Right prog -> do
-        let t = unDeBruijnAnonTerm $ UPLC.toTerm prog
-        return $! force t
-        -- `force` to try to ensure that deserialiation is not included in benchmarking time.
+readFlat :: FilePath -> IO BSL.ByteString
+readFlat file = BSL.fromStrict <$> BS.readFile file
 
-mkCekBM :: Term -> Benchmarkable
-mkCekBM program = whnf (UPLC.unsafeEvaluateCekNoEmit PLC.defaultCekParameters) program
+mkFullBM :: String -> BSL.ByteString -> Benchmarkable
+mkFullBM file = whnf (UPLC.unsafeEvaluateCekNoEmit PLC.defaultCekParameters
+                      . unDeBruijnAnonTerm
+                      . unsafeUnflat file
+                     )
 
-mkScriptBM :: FilePath -> FilePath -> Benchmark
-mkScriptBM dir file =
-    env (loadFlat $ dir </> file) $ \script -> bench (dropExtension file) $ mkCekBM script
-
--- Make benchmarks for the given files in the directory
-mkBMs :: FilePath -> [FilePath] -> [Benchmark]
-mkBMs dir files = map (mkScriptBM dir) files
+mkCekBM :: String -> BSL.ByteString -> Benchmarkable
+mkCekBM file program =
+    -- don't count the undebruijn . unflat cost
+    -- `force` to try to ensure that deserialiation is not included in benchmarking time.
+    let dbterm' = force (unDeBruijnAnonTerm $ unsafeUnflat file program)
```

Comment:
I think you probably need a bang pattern on the let too, `force` only deeply evaluates the argument if it itself is demanded.